File: examples/plugin-archives/demo-declarative-table/js/provider.js
function execute(params, upstream, context, api) {
  const rows = Math.max(1, Math.trunc(Number(params.rows ?? 6)));
  const scale = Number(params.scale ?? 2);
  const includeSquared = Boolean(params.includeSquared);
  return api.Table.fromRecords(Array.from({ length: rows }, (_, index) => {
    const value = index * scale;
    return includeSquared ? { index, value, squared: value * value } : { index, value };
  }));
}

File: examples/plugin-archives/demo-declarative-table/python/provider.py
def execute(params, upstream, context):
    rows = max(1, int(params.get("rows", 6)))
    scale = float(params.get("scale", 2))
    include_squared = bool(params.get("includeSquared", True))
    values = [index * scale for index in range(rows)]
    data = {"index": list(range(rows)), "value": values}
    if include_squared:
        data["squared"] = [value * value for value in values]
    return pd.DataFrame(data)
